src/phasorpy/phasor.py

```diff
@@ -2998,6 +3000,168 @@ def phasor_filter_median(
+    sigma : float, optional
+        Significance level to test the difference between two phasors, given
+        in terms of the equivalent 1D standard deviations. sigma=2 corresponds
+        to ~95% (or 5%) significance.
+    levels : int, optional
+        Number of levels for the wavelet decomposition.
+        Controls the maximum averaging area, which has a length of 2**level.
+    harmonic : int, sequence of int, or 'all', default: 1
+        Harmonics included in `real` and `imag`.
+        If an integer, the harmonics at which `real` and `imag` were acquired
+        or calculated.
+        If a sequence, the harmonics included in the first axis of `real` and
+        `imag`.
+        If `'all'`, the first axis of `real` and `imag` contains lower
+        harmonics.
+        The default is the first harmonic (fundamental frequency).
+    skip_axis : int or sequence of int, optional
+        Axes in `mean` to exclude from filter.
+        By default, all axes except harmonics are included.
+    normalize : bool, optional
+        By default (False), expects and returns normalized phasor coordinates.
+        If True, unnormalized phasor coordinates are expected as input.
+    samples : int, optional
+        Number of samples used to calculate the phasor coordinates.
+        By default, 1.
+    nan_safe : bool, optional
+        By default, distribute NaNs among filtered arrays.
+
+    Returns
+    -------
+    mean : ndarray
+        Unchanged intensity of phasor coordinates.
+    real : ndarray
+        Filtered real component of phasor coordinates.
+    imag : ndarray
+        Filtered imaginary component of phasor coordinates.
+
+    Raises
+    ------
+    ValueError
+        If `level` is less than 0.
+        If `samples` is less than 1.
+        If the number of harmonics in `harmonic` is less than 2.
+        If `real` and `imag` have no harmonic axis.
+        The array shapes of `mean`, `real`, and `imag` do not match.
+
+    Examples
+    --------
+
+    """
+    from pawflim import pawflim  # type: ignore[import-untyped]
+
+    mean = numpy.asarray(mean)
+    real = numpy.asarray(real)
+    imag = numpy.asarray(imag)
+
+    if levels < 0:
+        raise ValueError(f'{levels=} < 0')
+    if levels == 0:
+        return mean, real, imag
+    if samples < 1:
+        raise ValueError(f'{samples=} < 1')
+
+    if mean.shape != real.shape[-mean.ndim if mean.ndim else 1 :]:
+        raise ValueError(f'{mean.shape=} != {real.shape=}')
+    if real.shape != imag.shape:
+        raise ValueError(f'{real.shape=} != {imag.shape=}')
+
+    has_harmonic_axis = mean.ndim + 1 == real.ndim
+    if not has_harmonic_axis:
+        raise ValueError('No harmonic axis')
+    harmonic_max = real.shape[0]
+    if harmonic_max < 2:
+        raise ValueError(f'{harmonic=} < 2')
+    harmonics, _ = parse_harmonic(harmonic, harmonic_max)
+    harmonic_list = [
+        n for n in harmonics if 2 * n in range(1, harmonic_max + 1)
+    ]
```

Comment:
I don't think this is quite correct. The `harmonics` input parameter describes which harmonics are contained in the `real` and `imag` arrays (which may be, for example, 1 and 2, or 2 and 4, or 1 to 4), but here it is assumed that they always include 1 to `harmonic_max`.